src/ui/public/vislib/visualizations/column_chart.js

```diff
@@ -109,55 +98,52 @@ export default function ColumnChartFactory(Private) {
      */
     addStackedBars(bars) {
       const data = this.chartData;
-      const xScale = this.handler.categoryAxes[0].getScale();
-      const yScale = this.handler.valueAxes[0].getScale();
+      const xScale = this.getCategoryAxis().getScale();
+      const yScale = this.getValueAxis().getScale();
       const height = yScale.range()[0];
       const yMin = yScale.domain()[0];
 
       let barWidth;
-      if (data.ordered && data.ordered.date) {
-        const start = data.ordered.min;
-        const end = moment(data.ordered.min).add(data.ordered.interval).valueOf();
+      if (this.getCategoryAxis().config.isTimeDomain()) {
+        const ordered = this.handler.data.get('ordered');
+        const start = ordered.min;
+        const end = moment(ordered.min).add(ordered.interval).valueOf();
 
         barWidth = xScale(end) - xScale(start);
         barWidth = barWidth - Math.min(barWidth * 0.25, 15);
       }
 
       // update
       bars
-        .attr('x', function (d) {
-          return xScale(d.x);
-        })
-        .attr('width', function () {
-          return barWidth || xScale.rangeBand();
-        })
-        .attr('y', function (d) {
-          if (d.y < 0) {
-            return yScale(d.y0);
-          }
-
-          return yScale(d.y0 + d.y);
-        })
-        .attr('height', function (d) {
-          if (d.y < 0) {
-            return Math.abs(yScale(d.y0 + d.y) - yScale(d.y0));
-          }
-
-          // Due to an issue with D3 not returning zeros correctly when using
-          // an offset='expand', need to add conditional statement to handle zeros
-          // appropriately
-          if (d._input.y === 0) {
-            return 0;
-          }
-
-          // for split bars or for one series,
-          // last series will have d.y0 = 0
-          if (d.y0 === 0 && yMin > 0) {
-            return yScale(yMin) - yScale(d.y);
-          }
+      .attr('x', function (d) {
+        return xScale(d.x);
+      })
+      .attr('width', function () {
+        return barWidth || xScale.rangeBand();
+      })
+      .attr('y', function (d) {
+        return yScale(d.y0 + d.y);
+      })
+      .attr('height', function (d) {
+        if (d.y < 0) {
+          return Math.abs(yScale(d.y0 + d.y) - yScale(d.y0));
+        }
+        // todo:
+        // Due to an issue with D3 not returning zeros correctly when using
+        // an offset='expand', need to add conditional statement to handle zeros
```

Comment:
If we can't reproduce the issue this was fixing I would feel more comfortable leaving it for now. Subsequent cleanup prs could address stuff like this?